### DISCUSSION TITLE
Server Setup Sluggish Performance

### DISCUSSION THREAD

**Discussion by allw1994** [mastodon/mastodon#20272] [Q&A]
Trying to set up a server but it almost constantly shows either a something went wrong page or is unbelievably slow, currently the feed webpage has loaded but after 10 minutes it has still not loaded the contents of the feed. Sidekiq shows that there are above 16,000 jobs to process but it's not even doing 1 job a second hence why I have increased the jobs/worker processes limit. As I write this Sidekiq is also showing that it is idle but there are still a massive number of jobs waiting.

The relevant bits of `.env.production`:
````
SINGLE_USER_MODE=false
DB_POOL=50
REDIS_HOST=localhost
ES_ENABLED=true
ES_HOST=localhost
MAX_THREADS=10
WEB_CONCURRENCY=25
STREAMING_CLUSTER_NUM=20
````
Additionally, sidekick has 30 concurrent jobs, of which 29 are busy at the time of writing. System stats show near-constant 90% Memory and around 10% CPU usage, the swap file is at 10% capacity or less. The DB is hosted on a different machine.

I also suspect that parts of Mastodon crash after several hours, hence stopping the website from loading properly. Unless I increase the timeout values on Nginx I would regularly get timeout errors.

Currently, there are only 2 users (both me) and around 5 posts.  Though the server is subscribed to 3 relays which would explain the large pull queue in Sidekiq.

**Comment by draft13** [comment]
I have a similar situation too. An instance with 3 users but the timeline(s) are so slow to load. Not sure how to diagnose.
